server/service/ms/etcd/etcd.go

```diff
@@ -185,3 +341,219 @@ func (ds *DataSource) UpdateRule() {
 func (ds *DataSource) DeleteRule() {
 	panic("implement me")
 }
+
+func (ds *DataSource) modifySchemas(ctx context.Context, domainProject string, service *pb.MicroService,
+	schemas []*pb.Schema) *scerr.Error {
+	remoteIP := util.GetIPFromContext(ctx)
+	serviceID := service.ServiceId
+	schemasFromDatabase, err := getSchemasFromDatabase(ctx, domainProject, serviceID)
+	if err != nil {
+		log.Errorf(nil, "modify service[%s] schemas failed, get schemas failed, operator: %s",
+			serviceID, remoteIP)
```

Comment:
this func is too long， and cyclo is high  @little-cui could u please refactor this func